gsd/hoomd.py

```diff
@@ -1092,3 +1097,44 @@ def open(name, mode='rb'):
                          schema_version=[1, 4])
 
     return HOOMDTrajectory(gsdfileobj)
+
+def read_log(name):
+    """Read scalar values from a hoomd schema GSD file.
+
+    Args:
+        name (str): File name to open.
+
+    Returns:
+        `pandas.DataFrame`
+    """
+    if fl is None:
+        raise RuntimeError("file layer module is not available")
+    if gsd is None:
+        raise RuntimeError("gsd module is not available")
+    if pandas is None:
+        raise RuntimeError("pandas module is not available")
+
+    gsdfileobj = fl.open(name=str(name),
+                         mode='rb',
+                         application='gsd.hoomd ' + gsd.__version__,
+                         schema='hoomd',
+                         schema_version=[1, 4])
+
+    logged_data_names = gsdfileobj.find_matching_chunk_names('log/')
+    if len(logged_data_names) > 0:
+        logged_data_dict = dict()
+        for log in logged_data_names:
+            if gsdfileobj.chunk_exists(frame=0, name=log):
+                tmp = gsdfileobj.read_chunk(frame=0, name=log)
+                if tmp.shape[0] == 1:
+                    logged_data_dict[log] = [*tmp]
+
+        for idx in range(1, gsdfileobj.nframes):
+            for log in logged_data_dict.keys():
+                if gsdfileobj.chunk_exists(frame=idx, name=log):
+                    logged_data_dict[log].extend(gsdfileobj.read_chunk(frame=idx, name=log))
```

Comment:
```suggestion
                if gsdfileobj.chunk_exists(frame=idx, name=log):
                    logged_data_dict[log].extend(gsdfileobj.read_chunk(frame=idx, name=log))
                else:
                    # read the log data from frame 0
```
If logged data is missing from a frame, it is assumed to be constant with the value at frame 0: https://gsd.readthedocs.io/en/v2.7.0/schema-hoomd.html#logged-data